### PR TITLE
Vb/#586 grid io it

### LINKED ISSUE #586: Provide integration test for grid io
With the current issue #582 it became apparent that sinks and sources for PSDM data are not necessarily always compatible, i.e. a sink producing output that cannot be read by a corresponding source.

To further investigate such cases and prevent them from happening in the future, an integration test should be provided that writes out an example grid, reads it back in (or vice-versa) and compares results.

### PR DESCRIPTION
**Pull request by vickybung1**
Resolves #586

### CODE REVIEW

**Comment by vickybung1** [comment]
#710 must be solved first

**Comment by vickybung1** [comment]
Additionally resolves #502

**Comment by sebastian-peter** [comment]
!test

File: src/test/groovy/edu/ie3/datamodel/io/GridIoIT.groovy [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+class GridIoIT extends Specification implements CsvTestDataMeta {
+
+  def "Input JointGridContainer equals Output JointGridContainer."(){
+
+    given:
+    // create joint grid container
+    def gridname = new String("vn_simona")
+    def seperator = new String(",")
+    def folderpath = new String(jointGridFolderPath)
```
Comment (sebastian-peter) [inline src/test/groovy/edu/ie3/datamodel/io/GridIoIT.groovy:23]: You could use the String `jointGridFolderPath` directly below

File: src/test/groovy/edu/ie3/datamodel/io/GridIoIT.groovy [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    // write files from joint grid container in output directory
+    sink.persistJointGrid(firstGridContainer)
+
+    // create second grid container from output folder
+    def secondGridContainer = CsvJointGridContainerSource.read(gridname, seperator, outDirectoryPath)
+
+    // delete files in output directory
+    tempDirectory.toFile().eachFile {
+      it.deleteOnExit()
+    }
```
Comment (sebastian-peter) [inline src/test/groovy/edu/ie3/datamodel/io/GridIoIT.groovy:41]: This you could put into a `cleanupSpec()`, similar to how it works in `CsvFileConnectorTest`

File: src/test/groovy/edu/ie3/datamodel/io/GridIoIT.groovy [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    then:
+    //compare input and output joint grid container
+
+    firstGridContainer.equals(secondGridContainer)
```
Comment (sebastian-peter) [inline src/test/groovy/edu/ie3/datamodel/io/GridIoIT.groovy:46]: Could be replaced by `firstGridContainer == secondGridContainer` (this works in Groovy)

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
Some minor remarks from my side

File: src/test/groovy/edu/ie3/datamodel/io/GridIoIT.groovy (hunk 2 above)
Comment (sebastian-peter) [reply src/test/groovy/edu/ie3/datamodel/io/GridIoIT.groovy:41]: Also I just realized this doesn't work and the files are not deleted from my temp folder. Maybe you could look into this

**Comment by sebastian-peter** [comment]
Note to myself: we should use the updated csv sources from https://github.com/ie3-institute/simona/pull/424 once they are final.

File: src/test/groovy/edu/ie3/datamodel/io/GridIoIT.groovy (hunk 2 above)
Comment (sebastian-peter) [reply src/test/groovy/edu/ie3/datamodel/io/GridIoIT.groovy]: I just swapped two lines of code and it works for me now... Please check for yourself also.

**Review by sebastian-peter — APPROVED** [review]
Ok I only made some minor final adaptions. Let's get this merged!

**Pull request merged**